inference-engine/src/cldnn_engine/ops/loop.cpp

```diff
@@ -0,0 +1,250 @@
+    auto mem = p.GetEngine().allocate_memory(output_layout);
+    auto md = cldnn::mutable_data(id, {input}, mem); // cldnn::data cannot set dependency
+    return md;
+}
+
+static void UpdateBackedge(std::vector<cldnn::loop::backedge_mapping>& back_edges,
+                            const cldnn::primitive_id& old_primitive_id, const cldnn::primitive_id& new_primitive_id) {
+    for (auto& back_edge : back_edges) {
+        if (back_edge.from == old_primitive_id) {
+            back_edge.from = new_primitive_id;
+        }
+    }
+}
+
+static std::string GetExternalInputName(const int64_t body_parameter_index,
+                                        const std::shared_ptr<Loop>& op) {
+    const auto& loop_input_descs = op->get_input_descriptions();
+    for (const auto& loop_input_desc : loop_input_descs) {
+        if (loop_input_desc->m_body_parameter_index == body_parameter_index) {
+            auto external_node = op->get_input_node_shared_ptr(loop_input_desc->m_input_index);
+            return layer_type_name_ID(external_node);
+        }
+    }
+    return {""};
+}
+
+static InferenceEngine::Precision GetPrecisionFromNgraphType(ngraph::element::Type_t type) {
+#define CASE(ngraph_type, ie_type) \
+    case ngraph::element::ngraph_type: \
+        return InferenceEngine::Precision::ePrecision::ie_type;
+    switch (type) {
+        CASE(i8, I8)
+        CASE(i16, I16)
+        CASE(i32, I32)
+        CASE(i64, I64)
+        CASE(u8, U8)
+        CASE(u16, U16)
+        CASE(u32, U32)
+        CASE(u64, U64)
+    }
+#undef CASE
+    throw std::runtime_error("not integer type");
+    return {};
+}
+
+void CreateLoopOp(Program& p, const std::shared_ptr<Loop>& op) {
+    const std::string layerName = layer_type_name_ID(op);
+    auto inputPrimitives = p.GetInputPrimitiveIDs(op);
+    const auto& loop_input_descs = op->get_input_descriptions();
+    const auto& loop_output_descs = op->get_output_descriptions();
+    const auto& body_inputs = op->get_function()->get_parameters();
+    const auto& body_outputs = op->get_function()->get_results();
+
+    InferenceEngine::CNNNetwork body_network(op->get_function());
+    auto networkInputs = body_network.getInputsInfo();
+    auto networkOutputs = body_network.getOutputsInfo();
+
+    // Set special body ports: current_iteration input , execution condition output
+    auto special_body_ports = op->get_special_body_ports();
+
+    std::string body_current_iteration_id;
+    if (special_body_ports.current_iteration_input_idx >= 0) {
+        auto current_iteration_input = body_inputs.at(special_body_ports.current_iteration_input_idx);
+        body_current_iteration_id = layer_type_name_ID(current_iteration_input);
+        std::string input_name = ngraph::op::util::create_ie_output_name(current_iteration_input);
+        const auto networkInput = networkInputs.at(input_name);
+        if (networkInput->getPrecision().is_float()) {
```

Comment:
Why you change precision for floats only? Can `setPrecision` be called for all inputs?

Reply:
Remove float checking condition. please review it again. thanks.